File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/FantasyPriceDriver.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "fantasy_price_driver")
@Data
public class FantasyPriceDriver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "driver_id", referencedColumnName = "id")
    private Driver driver;

    @ManyToOne
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

    @Column(name = "price", nullable = false)
    private int price;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/fantasyPriceDriver.ts
import { Driver } from "./driver";
import { Race } from "./race";
import { Season } from "./season";

export class FantasyPriceDriver {
  id: number;
  driver: Driver;
  race: Race;
  price: number;
  season: Season;
  totalPoints: number = 0;
  averagePoints: number = 0;
  differencePrice?: number;

  constructor(id: number, driver: Driver, race: Race, price: number, season: Season) {
    this.id = id;
    this.driver = driver;
    this.race = race;
    this.price = price;
    this.season = season;
  }
}
